### ISSUE TITLE
Bug with german umlauts

### ISSUE THREAD

**Issue by Bio7** [r-lib/styler#374]
Today I discovered the package styler as an alternative to the formatR package.

However as I made some tests I discovered by chance a bug with german umlauts which leads to a cut off at the
position where an umlaut occur. Try to format the following R file (cut off at "Ärger")

For example:
```
# Test
plant <- setRefClass("plant",
    fields = list(name = "character", weight = "numeric", age = "numeric"),
    methods = list(
      getName = function(x) {
        name <<- name
      },
      getWeight = function(x) {
        weight <<- weight
      },
      getAge = function(x) {
        age <<- age
      }
    )
)
# Here the Ärger!
s <- plant(name = "Corynephorus canescens", weight = 15, age = 3.5)
```

The test was made on Windows as the OS using the style_file() command.
Windows text file encoding: CP1252

By the way is it possible to suppress the information which are printed to the console?

**Comment by lorenzwalthert** [comment]
Hi @Bio7, thanks for reporting this. Are you using styler 1.0.1? A related bug was fixed there (#329).

**Comment by Bio7** [comment]
sessionInfo() gives me: styler_1.0.1
Installed from the cran repo.

**Comment by Bio7** [comment]
R version 3.4.2 (2017-09-28)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 16299)

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
[3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] styler_1.0.1 Rserve_1.7-3

**Comment by lorenzwalthert** [comment]
Ok. I am on macOs, so it's not easy for me to reproduce that. However, styler is designed to work with UTF8 encoded files. If you use RStudio, can you save the file with encoding "UTF-8" and tell me whether it works?

**Comment by lorenzwalthert** [comment]
I c/p your code, changed to your encoding to CP1252 and I can reproduce your problem.

**Comment by Bio7** [comment]
I suspected that the file encoding could be the issue. By changing my default file encoding the formatting works.
However a lot of text editors and IDE's are using the default Windows OS system encoding by default which could cause a lot of headaches and corrupted files. So for windows this should be handled.

**Comment by lorenzwalthert** [comment]
I can read in the file with `enc::read_lines_enc(..., file_encoding = "latin1")` correctly but I think as CP1252 is a superset of "latin1", there are edge cases with incorrect behaviour.
@krlmlr do you think we should support more file encodings? Is the package `enc` limited to `utf8` and `latin1`? If not, we could add an argument `file_encoding` to the exported `style_*()` functions. As mentioned above, `formatR` seems to handle the cases well.

**Comment by Bio7** [comment]
Else the formatting works very fine especially on S4, Ref. and R6 classes whereas formatR fails (also on inline comments). Thank you for this alternative.

Apropos to my question:

Is it possible to suppress the printed R console information from the style_file() function ?

**Comment by lorenzwalthert** [comment]
Thanks, glad to hear that. It is not (yet) possible. As you probably know, you can always wrap code around a `capture.output()` and throw it away. I opened #375 to discuss this.

```r
silent_style_file <- function(...) {
  msg <- capture.output(style_file(...))

}
```

**Comment by lorenzwalthert** [comment]
I also opened #376 to discuss the support of file encodings.

**Comment by lorenzwalthert** [comment]
Closed in favour of #376.
